code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -2977,3 +2985,256 @@ func (srh *StorageRestHandler) replicateUserAggregates(w http.ResponseWriter, r
 	}
 	common.Respond(w, r, users, nil)
 }
+
+func (srh *StorageRestHandler) rewardProviders(w http.ResponseWriter, r *http.Request) {
+	// read all data from reward_providers table and return
+	edb := srh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	challengeID := r.URL.Query().Get("challenge_id")
+	providerID := r.URL.Query().Get("provider_id")
+	query := r.URL.Query().Get("query")
+
+	var rps []event.RewardProvider
+
+	if query != "" {
+		rps = edb.RunWhereQueryInProviderRewards(query)
+	} else if challengeID != "" {
+		rps = edb.GetChallengeRewardsByChallengeID(challengeID)
+	} else if providerID != "" {
+		rps = edb.GetChallengeRewardsByProviderID(providerID)
+	} else {
+		rps = edb.GetAllProviderChallengeRewards()
+	}
+
+	var sum int64
+
+	for _, rp := range rps {
+		cur, _ := rp.Amount.Int64()
+		sum += cur
+	}
+
+	result := map[string]interface{}{
+		"sum": sum,
+		"rps": rps,
+	}
+
+	// return the list of reward providers
+	common.Respond(w, r, result, nil)
+}
+
+func (srh *StorageRestHandler) rewardDelegates(w http.ResponseWriter, r *http.Request) {
+	// read all data from reward_delegates table and return
+	edb := srh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	challengeID := r.URL.Query().Get("challenge_id")
+	query := r.URL.Query().Get("query")
```

Comment:
why do we need run raw queries from clients? it's a security hole

Reply:
Added when debugging. We don't need this code now. Removed it.